homeassistant/components/foscam/camera.py

```diff
@@ -90,85 +87,92 @@
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up a Foscam IP Camera."""
+    LOGGER.warning(
+        "Loading foscam via platform config is deprecated, it will be automatically imported. Please remove it afterwards."
+    )
 
-    async def async_handle_ptz(service):
-        """Handle PTZ service call."""
-        movement = service.data[ATTR_MOVEMENT]
-        travel_time = service.data[ATTR_TRAVELTIME]
-        entity_ids = await async_extract_entity_ids(hass, service)
-
-        if not entity_ids:
-            return
-
-        _LOGGER.debug("Moving '%s' camera(s): %s", movement, entity_ids)
+    config_new = {
+        CONF_NAME: config[CONF_NAME],
+        CONF_HOST: config["ip"],
+        CONF_PORT: config[CONF_PORT],
+        CONF_USERNAME: config[CONF_USERNAME],
+        CONF_PASSWORD: config[CONF_PASSWORD],
+        CONF_STREAM: "Main",
+    }
 
-        all_cameras = hass.data[FOSCAM_DATA][FOSCAM_ENTITIES]
-        target_cameras = [
-            camera for camera in all_cameras if camera.entity_id in entity_ids
-        ]
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config_new
+        )
+    )
 
-        for camera in target_cameras:
-            await camera.async_perform_ptz(movement, travel_time)
 
-    hass.services.async_register(
-        FOSCAM_DOMAIN, SERVICE_PTZ, async_handle_ptz, schema=SERVICE_PTZ_SCHEMA
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a Foscam IP camera from a config entry."""
+    platform = entity_platform.current_platform.get()
+    platform.async_register_entity_service(
+        SERVICE_PTZ, SERVICE_PTZ_SCHEMA, "async_perform_ptz"
     )
 
+    config = {CONF_NAME: config_entry.title, **config_entry.data}
+
     camera = FoscamCamera(
-        config[CONF_IP],
+        config[CONF_HOST],
         config[CONF_PORT],
         config[CONF_USERNAME],
         config[CONF_PASSWORD],
```

Comment:
You could just use config_entry.data here
```suggestion
        config_entry.data[CONF_HOST],
        config_entry.data[CONF_PORT],
        config_entry.data[CONF_USERNAME],
        config_entry.data[CONF_PASSWORD],
```